### PR TITLE
[PSA-5] Make the BNO a member

### PR DESCRIPTION
**Pull request by aelahi23**
Circumvent the flow session.
BNO can still revoke and amend memberships.

### CODE REVIEW

File: bn-apps/memberships-management/membership-service/src/main/kotlin/com/r3/businessnetworks/membership/flows/bno/SelfIssueMembershipFlow.kt [hunk 1]
```diff
@@ -2,60 +2,100 @@ package com.r3.businessnetworks.membership.flows.bno
...
 /**
- * Self Issue a Membership.  This can only be done by a BNO
+ * Self Issue a Membership.  This can only be done by a BNO,
+ * @param bnoMembership bno membership state to be activated
  */
+@Suppress("UNUSED_VARIABLE")
 @StartableByRPC
 @InitiatingFlow(version = 2)
-open class SelfIssueMembershipFlow(val metaData : Any) : FlowLogic<SignedTransaction>() {
+open class SelfIssueMembershipFlow(val bnoMembership: StateAndRef<MembershipState<Any>>) : FlowLogic<SignedTransaction>() {
```
Comment (ivanr3) [inline bn-apps/memberships-management/membership-service/src/main/kotlin/com/r3/businessnetworks/membership/flows/bno/SelfIssueMembershipFlow.kt]: Hi Arif. As far as I understand from the class name, this flow is supposed to issue a membership onto the ledger, right? Why is it taking an *existing* membership as a parameter then?

Comment (aelahi23) [reply bn-apps/memberships-management/membership-service/src/main/kotlin/com/r3/businessnetworks/membership/flows/bno/SelfIssueMembershipFlow.kt]: The parameter is just for the BNO to enter their Identity so I can use it to first check if they are in the list of whitelisted BNO's and then change the BNO status to Active member of the network

Comment (ivanr3) [reply bn-apps/memberships-management/membership-service/src/main/kotlin/com/r3/businessnetworks/membership/flows/bno/SelfIssueMembershipFlow.kt]: Yes, but *before* activating a membership, you'd need to issue it first, right? How can BNO do that?

Comment (aelahi23) [reply bn-apps/memberships-management/membership-service/src/main/kotlin/com/r3/businessnetworks/membership/flows/bno/SelfIssueMembershipFlow.kt]: From my understanding the BNO node is created from the MembershipState already and this workflow was supposed to just change the status parameter of the BNO node to active

Comment (aelahi23) [reply bn-apps/memberships-management/membership-service/src/main/kotlin/com/r3/businessnetworks/membership/flows/bno/SelfIssueMembershipFlow.kt]: If the BNO is just a party I will need to create a membership state for it?

Comment (ivanr3) [reply bn-apps/memberships-management/membership-service/src/main/kotlin/com/r3/businessnetworks/membership/flows/bno/SelfIssueMembershipFlow.kt]: Yes BNO needs to be issued with a membership state (just as any other participant on the network). The lifecycle of membership for BNO is exactly the same as for everyone else, i.e. Issue -> Activate -> Revoke. The difference is only that for BNO, both member and BNO fields would be the same.

**Pull request closed without merging**